rclpy/test/test_node.py

```diff
@@ -838,6 +842,127 @@ def test_node_set_parameters_rejection(self):
+            ('reject', True, ParameterDescriptor()),
+            ('non_reject', True, ParameterDescriptor()),
+            ('non_reject', True, ParameterDescriptor())
+        ]
+        self.node.declare_parameters('', reject_list_parameter_tuple)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_1)
+
+        result = self.node.set_parameters(
+            [
+                Parameter(
+                    name=reject_list_parameter_tuple[0][0],
+                    value=reject_list_parameter_tuple[0][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple[1][0],
+                    value=reject_list_parameter_tuple[1][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple[2][0],
+                    value=reject_list_parameter_tuple[2][1]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
+        self.assertFalse(result[0].successful)
+        # Testing with setting new parameter with successful result with
+        # Previously declared function.
+        self.test_node_set_parameters()
+
+    def test_node_add_on_set_parameter_callback(self):
+        # Add callbacks to the list of callbacks.
+        callbacks = [
+            self.reject_parameter_callback,
+            self.reject_parameter_callback_1
+        ]
+        # for callback in callbacks:
+        #     self.node.add_on_set_parameters_callback(callback)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_1)
+
+        for callback in callbacks:
+            self.assertTrue(callback in self.node._parameters_callbacks)
+        for callback in callbacks:
+            self.node.remove_on_set_parameters_callback(callback)
+
+        # Adding the parameters which will be accepted without any rejections.
+        non_reject_parameter_tuple = [
+            ('N', True, ParameterDescriptor()),
+            ('N', True, ParameterDescriptor())
+        ]
+        self.node.declare_parameters('', non_reject_parameter_tuple)
+        
+        for callback in callbacks:
+            self.node.add_on_set_parameters_callback(callback)
+
+        result = self.node.set_parameters(
+            [
+                Parameter(
+                    name=non_reject_parameter_tuple[0][0],
+                    value=non_reject_parameter_tuple[0][1]
+                ),
+                Parameter(
+                    name=non_reject_parameter_tuple[1][0],
+                    value=non_reject_parameter_tuple[1][1]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
```

Comment:
We set two parameters, so we should check two results. Add more asserts for the second result:

```py
self.assertIsInstance(result[1], SetParametersResult)
self.assertTrue(result[1].successful)
```